[master] AWS SESSION TOKEN returned

Tanmoy037 opened the pull request:
### What does this PR do?
Fixes salt-cloud using ec2  authenticate fails
### What issues does this PR fix or reference?
Fixes: #64533

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [*] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

Ch3LL commented on changelog/64533.fixed.md:
```diff
@@ -0,0 +1,3 @@
+Added support for AWS session tokens in salt-cloud.
```
This is adding a feature, so the file extension needs to be `added` instead of fixed.

Ch3LL reviewed (changes requested):
This will also require test coverage

Ch3LL reviewed (dismissed):
Changelog looks good, this just needs test coverage now.

dwoz commented:
@Tanmoy037 Can you rebase this and resolve the conflicts? Ideally, add a test too.

twangboy reviewed (changes requested):
Please create a test for this

dwoz committed:
Add AWS session token support to salt-cloud EC2 provider

When a ``token`` key is present in the EC2 provider configuration,
forward it as the security token in AWS API calls. Previously only
the ``id`` and ``key`` were used for static credentials.

Fixes #64533
